kratos/includes/mesh.h

```diff
@@ -271,13 +271,29 @@ class Mesh : public DataValueContainer, public Flags
     /** Returns the Node::Pointer  corresponding to it's identifier */
     typename NodeType::Pointer pGetNode(IndexType NodeId)
     {
-        return (*mpNodes)(NodeId);
+        KRATOS_TRY
+
+        auto i = mpNodes->find(NodeId);
+        if (i == mpNodes->end())
+            KRATOS_THROW_ERROR(std::out_of_range, "node index not found: ", NodeId)
```

Comment:
I would suggest you to use the new KRATOS_ERROR macro:
`KRATOS_ERROR << "node index not found: " << NodeId;`